File: app/src/main/java/com/example/hspcadmin/htmlproject/activity/presenter/AbstractPresenter.java
package com.example.hspcadmin.htmlproject.activity.presenter;

import com.example.hspcadmin.htmlproject.activity.abstracts.ViewContractType;

/**
 * Created by hspcadmin on 2018/11/7.
 */

public class AbstractPresenter {
    ViewContractType.ViewType viewType;

    public AbstractPresenter(ViewContractType.ViewType viewType){
        this.viewType = viewType;
        viewType.initView();
    }


}

File: app/src/main/java/com/example/hspcadmin/htmlproject/activity/presenter/AbstractPresenter.kt
package com.example.hspcadmin.htmlproject.activity.presenter

import android.view.View
import com.example.hspcadmin.htmlproject.activity.abstracts.ViewContractType

/**
 * Created by hspcadmin on 2018/11/22.
 */
abstract class AbstractPresenter:ViewContractType.ViewAction{
    var mView:View? =null

    override fun initView(view: View?) {
        this.mView = view
    }

    override fun setAction() {
    }

}